src/util/OAuth2Util.js

```diff
@@ -95,6 +95,43 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
     }
   };
 
+  // Parse through the OAuth 'authParams' object to ensure the 'openid' scope is
+  // included (if required)
+  util.scrubScopes = function (authParams) {
+    if (!authParams.responseType) {
+      return;
+    }
+
+    //  Convert scope into an Array
+    var scope = Array.isArray(authParams.scopes) ? authParams.scopes : authParams.scopes.split(' ');
+
+    // Remove the 'openid' scope, as it is only required if an 'id_token' is requested
+    if (scope.indexOf('openid') > -1) {
+      scope.splice(scope.indexOf('openid'), 1);
+    }
+
+    // Add the 'openid' scope
+    if (authParams.responseType.indexOf('id_token') !== -1) {
+      scope.push('openid');
+    }
+
+    return scope;
+  };
+
+  // Utility handlers for maping convenience keys to OAuth params
+  util.getResponseType = function (options) {
```

Comment:
Ditto on "does this need to be it's own method"?  We have a fairly concrete set of logic to build up the necessary render call, given the inputs to `showSignInToGetTokens`, yet I feel like I'm having to chase down the logic across multiple files.

Reply:
`getReponseType` and `addOrRemoveOpenIdScope` (renamed) are going to be used with every new OIDC signIn method we provide. For example, the silent login (iframe) flow is a future task.